entraid/src/main/java/redis/clients/authentication/entraid/EntraIDIdentityProvider.java

```diff
@@ -0,0 +1,94 @@
+package redis.clients.authentication.entraid;
+
+import java.net.MalformedURLException;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+
+import com.microsoft.aad.msal4j.ClientCredentialFactory;
+import com.microsoft.aad.msal4j.ClientCredentialParameters;
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.IClientCredential;
+import com.microsoft.aad.msal4j.ManagedIdentityApplication;
+import com.microsoft.aad.msal4j.ManagedIdentityParameters;
+import redis.clients.authentication.core.IdentityProvider;
+import redis.clients.authentication.core.Token;
+
+public final class EntraIDIdentityProvider implements IdentityProvider {
+
+    private Supplier<IAuthenticationResult> resultSupplier;
+
+    public EntraIDIdentityProvider(ServicePrincipalInfo servicePrincipalInfo, Set<String> scopes,
```

Comment:
Its generally considered a bad idea to add logic in the constructor that could fail its creation.
An exception to this is classes such as URI where the contractor fails if the URI is malformed.

However here we don't validate the input parameters, but instead build up an app that connects remotely and there is a  potential that many of the steps could fail.

IMHO this needs to be either a separate step 
```java
provider.initialize();
```
or a lazy-loaded step to any of the public merhods (

```java
if(!initialized) { 
   initialize(); 
}
```

Reply:
Ali, https://stackoverflow.com/questions/6086334/is-it-good-practice-to-make-the-constructor-throw-an-exception this opinion may not be shared by the community, so please accept this only if you agree with me.

Reply:
it s ok; i think what you mentioned is especially valid for static ctors,, i personally had some bad experience with static constructors in the past and i got goosebumps whenever i see them around. Though this is not the case, only down side of moving it to a later stage defers the initiation of identity provider context, which is ok i think.